opentelemetry-etw-logs/src/logs/processor.rs

```diff
@@ -0,0 +1,302 @@
+    /// Creates a new instance of [`ProcessorBuilder`] with the given provider name.
+    ///
+    /// The provider name must be:
+    ///
+    /// - non-empty,
+    /// - less than 234 characters long, and
+    /// - contain only ASCII alphanumeric characters, underscores (`_`), or hyphens (`-`).
+    ///
+    /// At the same time, it is recommended to use a provider name that is:
+    /// - short
+    /// - human-readable
+    /// - unique
+    /// - describing the application or service that is generating the logs
+    ///
+    /// By default, all events will be exported to the "Log" ETW event. See [`ProcessorBuilder`] for details on how to configure a different behavior.
+    pub fn builder(provider_name: &str) -> ProcessorBuilder {
+        ProcessorBuilder::new(provider_name)
+    }
+
+    /// Creates a new instance of the [`Processor`] using the given options.
+    pub(crate) fn new(options: Options) -> Self {
+        let exporter: ETWExporter = ETWExporter::new(options);
+        Processor {
+            event_exporter: exporter,
+        }
+    }
+}
+
+impl opentelemetry_sdk::logs::LogProcessor for Processor {
+    fn emit(&self, data: &mut SdkLogRecord, instrumentation: &InstrumentationScope) {
+        let log_tuple = &[(data as &SdkLogRecord, instrumentation)];
+        // TODO: How to log if export() returns Err? Maybe a metric? or eprintln?
+        let _ = futures_executor::block_on(self.event_exporter.export(LogBatch::new(log_tuple)));
+    }
+
+    // Nothing to flush as this processor does not buffer
+    fn force_flush(&self) -> OTelSdkResult {
+        Ok(())
+    }
+
+    fn shutdown(&self) -> OTelSdkResult {
+        self.event_exporter.shutdown()
+    }
+
+    #[cfg(feature = "spec_unstable_logs_enabled")]
+    fn event_enabled(
+        &self,
+        level: opentelemetry::logs::Severity,
+        target: &str,
+        name: Option<&str>,
+    ) -> bool {
+        use opentelemetry_sdk::logs::LogExporter;
+
+        self.event_exporter.event_enabled(level, target, name)
+    }
+
+    fn set_resource(&mut self, resource: &Resource) {
+        self.event_exporter.set_resource(resource);
+    }
+}
+
+/// Builder for configuring and constructing a [`Processor`].
+#[derive(Debug)]
+pub struct ProcessorBuilder {
+    options_builder: Options,
+}
+
+impl ProcessorBuilder {
+    /// Creates a new instance of [`ProcessorBuilder`] with the given provider name.
+    ///
+    /// By default, all events will be exported to the "Log" ETW event.
+    pub(crate) fn new(provider_name: &str) -> Self {
+        ProcessorBuilder {
+            options_builder: Options::new(provider_name.to_string()),
+        }
+    }
+
+    /// Sets a user-defined callback that returns the ETW event name, using the the [`SdkLogRecord`] as input.
+    ///
+    /// The resulting name must be a valid CommonSchema 4.0 TraceLoggingDynamic event name. Otherwise,
+    /// the default "Log" ETW event name will be used.
+    pub fn etw_event_name_from_callback(
```

Comment:
Please see the discussion here : https://github.com/open-telemetry/opentelemetry-rust-contrib/pull/275#discussion_r2103713620

We need to decide how best to handle this.